x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteForecastAction.java

```diff
@@ -105,17 +105,19 @@ protected void doExecute(Task task, DeleteForecastAction.Request request, Action
             e -> handleFailure(e, request, listener)
         );
 
-        SearchSourceBuilder source = new SearchSourceBuilder();
-
-        BoolQueryBuilder builder = QueryBuilders.boolQuery()
-            .filter(QueryBuilders.termQuery(Result.RESULT_TYPE.getPreferredName(), ForecastRequestStats.RESULT_TYPE_VALUE));
+        BoolQueryBuilder query =
+            QueryBuilders.boolQuery()
+                .filter(QueryBuilders.termQuery(Result.RESULT_TYPE.getPreferredName(), ForecastRequestStats.RESULT_TYPE_VALUE));
         QueryBuilderHelper
             .buildTokenFilterQuery(Forecast.FORECAST_ID.getPreferredName(), forecastIds)
-            .ifPresent(builder::filter);
-        source.query(builder);
-
-        SearchRequest searchRequest = new SearchRequest(AnomalyDetectorsIndex.jobResultsAliasedName(jobId));
-        searchRequest.source(source);
+            .ifPresent(query::filter);
+        SearchSourceBuilder source =
+            new SearchSourceBuilder()
+                .size(MAX_FORECAST_TO_SEARCH)
+                .query(query);
+        SearchRequest searchRequest =
```

Comment:
First we query to find the forecast IDs the the job then run a DBQ on that list of forecast IDs. There is an optimisation we can make here to no bring back entire document when only one field (`forecast_id`) is required. Now that the query size is `MAX_FORECAST_TO_SEARCH` instead of 10 not bringing back the entire document is even more important. 

Could you make that change in this PR please

Reply:
> There is an optimisation we can make here to no bring back entire document when only one field (forecast_id) is required.

3 fields are required:
- forecast_id
- job_id (for parsing)
- status (for validation in line 127

I made the  change to only fetch these 3 fields.
One (small?) issue with that is that we still use `ForecastRequestStats` as a data structure to pass these fields around.
It may bite us in the future if someone tries to access another field from `ForecastRequestStats` forgetting it's incomplete.